### PR TITLE
Add Style Guide to C++

### PR DESCRIPTION
**Pull request by Noah11012**
See the issue #614

### CODE REVIEW

**Comment by jrg94** [comment]
Sorry! Had a hackathon this weekend. I'll take a look at this shortly.

File: archive/c/c-plus-plus/README.md [hunk 1]
```diff
@@ -14,6 +14,15 @@ Welcome to Sample Programs in C++!
...
 - Typing: Static
 
+## Style Guide
+For a long time, the styling that you would use in your code would depend on the company you worked for. Google, Mozilla, Apple, etc have their own style guidelines that their employees must follow for a more consistent codebase. Even individual developers form their own styles.
+
+A set of guidelines has been created by Bjarne Stroustrup, the language creator of C++ and Herb Sutter, a well known C++ expert in the community. The document is currently under continuous development, but it still serves well as an excellent place to read up on the best coding practices for C++ projects.
+
+The guidelines are separated into sections that deals with a language topic such as classes and exceptions.
+
+Link: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines
```
Comment (jrg94) [inline archive/c/c-plus-plus/README.md]: Everything looks good to me, but I was wondering if you could link these more elegantly in a sentence? For instance, maybe you could link this directly in the second paragraph over `set of guidelines`.

**Comment by Noah11012** [comment]
I was starting to get a bit worried about your absence.

File: archive/c/c-plus-plus/README.md (hunk 1 above)
Comment (Noah11012) [reply archive/c/c-plus-plus/README.md]: Done

**Comment by jrg94** [comment]
Yeah! Well, life happens. haha I've been falling behind on articles too, so I'm hoping to get some free time soon to write again.

**Comment by jrg94** [comment]
Hey @Noah11012! I think I want to revisit this. Could we perhaps just link the style guide in the references section? I think I'm going to rework the contributing doc to enforce that for new languages.

I think the README should just include lists of information. For instance, the Python README has a syntax section. If we need to explain anything more in depth, we should be writing in depth articles.

**Comment by Noah11012** [comment]
I'll create a new pull request soon.

**Pull request merged**